### PR TITLE
client side data flow/ data structure, side effects solution

### PR DESCRIPTION
**Pull request by theJian**

### CODE REVIEW

File: src/store/actions/ERR.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const ERR = actionType = `${actionType}#ERR`;
```
Comment (shadeofgod) [inline src/store/actions/ERR.js]: 嘲笑你

File: src/service/methods.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { resolve, format } from 'url';
+
+const api = path => resolve('https://langchao.org/api/', path);
+
+export const get = (path, query = {}, options = {}) =>
+  fetch(format({
```
Comment (incrediblink) [inline src/service/methods.js]: fetch 是 redux-loop 自带的一个方法吗？

File: src/components/Events.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ const Events = ({ onEventPress }) => (
       </View>
     </View>
     <SectionList
-      sections={eventList}
+      sections={eventList.map(data => ({ data: [data] }))}
```
Comment (theJian) [inline src/components/Events.js]: Dirty code in here.
We can fix it later.

File: src/store/reducers/events.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import { getAll } from '../../service/events.js';
+
+export default combineReducers({
+  data: requestData(
+    fetchList.type,
+    getAll,
+    (_, res) => res.eventList,
+    (_, err) => {
+      // TODO: handle error
```
Comment (theJian) [inline src/store/reducers/events.js]: Not implemented yet.
Need to know how to show the error message properly.

Comment (alanzchen) [reply src/store/reducers/events.js]: A modal？

**Comment by theJian** [comment]
Smash the merge button and we're so ready to build a fully functional app with the robust foundation(ignoring the fact that we haven't written any test, for now).

**Comment by alanzchen** [comment]
Blank screen on "Today" tab view (https://expo.io/@v2land/v2land-mobile?release-channel=feat-loop)

**Pull request merged**